eth/downloader/downloader.go

```diff
@@ -741,9 +752,36 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		}
 	}
 
+	// Only attempt span search if local head is "close" to reported remote
+	if remoteHeight-localHeight <= uint64(MaxHeaderFetch) {
+		a, err := d.findAncestorSpanSearch(p, remoteHeight, localHeight, floor)
+		if err == nil {
+			if a > localHeight {
+				a = localHeight
```

Comment:
Hm, what happens here? The found ancestor is above our local height? 
Ah, that's when we're aware of a higher block, but we don't consider it canon, for TD reasons...

Reply:
Yeap, I think you're correct. Fixed with https://github.com/ethereum/go-ethereum/pull/21063/commits/58dd36b66f6ab27e93527869a417e8edaab3dd31